plugin/edit.py

```diff
@@ -41,13 +41,21 @@ def open_and_apply_edits(self, path, file_changes):
 
 
 class LspApplyDocumentEditCommand(sublime_plugin.TextCommand):
+
     def run(self, edit, changes: 'Optional[List[TextEdit]]' = None):
         # Apply the changes in reverse, so that we don't invalidate the range
         # of any change that we haven't applied yet.
         if changes:
-            for change in sort_by_application_order(changes):
+            last_row, last_col = self.view.rowcol(self.view.size())
+            for change in reversed(sort_by_application_order(changes)):
                 start, end, newText = change
                 region = sublime.Region(self.view.text_point(*start), self.view.text_point(*end))
+
+                # Handle when a language server (eg gopls) inserts at a row beyond the document
+                # some editors create the line automatically, sublime needs to have the newline prepended.
+                if start[0] > last_row and newText[0] != '\n':
+                    newText = '\n' + newText
```

Comment:
Looks like an OK workaround.

Reply:
Thanks for the test you wrote and the research!